prow/git/git_command_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+)
+
+type fakeExecCommandExecutor struct {
+	ExpectedCommands []string // expected command lines' REGEX
+	CallNumber       int      // After execution, the number of times fakeExecCommand was called
+}
+
+func (f *fakeExecCommandExecutor) execCommand(command string, args ...string) *exec.Cmd {
+	cs := []string{"-test.run=TestHelperProcess", "--", command}
```

Comment:
This doesn't make any sense to me. Why is a test calling a test?

Reply:
This is how go authors are testing os/exec:
https://github.com/golang/go/blob/master/src/os/exec/exec_test.go#L33

Reply:
There's no need to test os.exec. I'm confident in the core golang libraries. I'm confident in the behavior of git.

I'm not confident of the arguments we send to os.exec. 

AKA the issue here is not that git interpreted the arguments we send it incorrectly, it is that we sent git bad arguments. We can fix that by validating the arguments we send git. This does not require subprocesses.

Reply:
@fejta agree. I can modify this. Would it be a throw away work considering this https://github.com/kubernetes/test-infra/pull/11248 ?

Reply:
Right now we're blocked from updating.

I agree with Steve that refactoring things to make them more testable is a good long-term solution.

For the immediate term, I would prioritize a small/medium fix that focuses on addressing the outage (on the other hand, I think we may have already rolled back the other PR?)

Reply:
It is rolled back. With functionality I am looking for (corp github). So I am definitely interested in including the rollback-rollback into this PR and fixing it.

Reply:
ok now it includes the rollback rollback, as well as reworked test.